src/main/java/org/apache/sling/resourceresolver/impl/mapping/ResourceMapperImpl.java

```diff
@@ -291,7 +292,7 @@ private List<String> readAliasesOptimized(String path) {
     	if ( aliases == null || !aliases.containsValue(name) ) 
     		return Collections.emptyList();
 
-    	return aliases.entrySet().stream()
+    	return new LinkedHashMap<>(aliases).entrySet().stream()
```

Comment:
Doesn't the `LinkedHashMap` constructor iterate over the `aliases` map? If that's the case nothing is solved.

Reply:
Indeed, I am looking into reproducing the concurrency issues in a unit test and afterwards come up with a better fix.